File: physicsCalc.c
/* Created by Tony Schneider on 1/29/2012
 * Last Modified by Tony Schneider on 1/29/2012
 *
 * physicsCalc.c
 * Used to determine the force of an object.
 */

#include <stdlib.h>
#include <stdio.h>
#include <math.h>
 
//TODO: place your function prototypes here
double getVelocity(double distance, double time);
double getAcceleration(double initVelocity, double velocity, double time);
double getForce(double mass, double acceleration); 
 
int main(int argc, char **argv)
{
    double distance, time, initVelocity, mass;

    printf("Enter the distance the object travled(m): ");
    scanf("%lf", &distance);
    
    printf("Enter the time it took to travel that distance(s): ");
    scanf("%lf", &time);

    printf("Enter the initial velocity of the object(m/s): ");
    scanf("%lf", &initVelocity);

    printf("Enter the mass of the object(g): ");
    scanf("%lf", &mass);

    //TODO: call your functions here
	double velocity = getVelocity(distance, time);
	double acceleration = getAcceleration(initVelocity, velocity, time);
	double force = getForce(mass, acceleration);
	
    //TODO: Print the results here
	printf("The velocity of the object is %lf m/s, the acceleration of the object is %lf m/s^2, and the total force of the object is %lf N.\n", velocity, acceleration, force);
	
    return 0;
}

//TODO: place your function definitions here
double getVelocity(double distance, double time) {
	double velocity = distance/time;
	return velocity;
}

double getAcceleration(double initVelocity, double velocity, double time) {
	double acceleration = (velocity - initVelocity)/time;
	return acceleration;
}

double getForce(double mass, double acceleration) {
	double force = mass*acceleration;
	return force;
}

File: PhysicsCalc.java
/**
 * PhysicsCalc.java
 * Carden Reichert
 * This program computes the force of an object.  The methods need to be
 * implemented.
 * 
 * Adapted from physicsCalc.c by Tony Schneider.
 * CSE 155H Fall 2012
 */
import java.util.Scanner;

public class PhysicsCalc {
    
	public static double getVelocity(double distance, double time) {
		double velocity = distance/time;
		return velocity;
	}

	public static double getAcceleration(double initVelocity, double velocity, double time) {
		double acceleration = (velocity - initVelocity)/time;
		return acceleration;
	}

	public static double getForce(double mass, double acceleration) {
		double force = mass*acceleration;
		return force;
	}
	
    public static void main( String[] args ){
        
        Scanner input = new Scanner( System.in );
        
        
        double distance, time, initVelocity, mass;
        
        System.out.print( "Enter the distance the object traveled: " );
        distance = input.nextDouble();
        System.out.println();
        
        System.out.print( "Enter the time it took to travel that distance: " );
        time = input.nextDouble();
        System.out.println();
        
        System.out.print( "Enter the initial velocity of the object(m/s): " );
        initVelocity = input.nextDouble();
        System.out.println();
        
        System.out.print( "Enter the mass of the object(g): " );
        mass = input.nextDouble();
        System.out.println();
        
        
        /* Invoke your written methods here */
        double velocity = getVelocity(distance, time);
		double acceleration = getAcceleration(initVelocity, velocity, time);
		double force = getForce(mass, acceleration);
        
        /* Print the result here */
        System.out.printf("The velocity of the object is %f m/s, the acceleration of the object is %f m/s^2, and the total force of the object is %f N.\n", velocity, acceleration, force);
		
    }
    
}
